src/sql/workbench/browser/modal/modal.ts

```diff
@@ -168,40 +189,76 @@ export abstract class Modal extends Disposable implements IThemable {
 
 	/**
 	 * Build and render the modal, will call {@link Modal#renderBody}
+	 *
 	 */
 	public render() {
+		let hasMacClass: boolean = document.querySelector('body.mac') !== null ? true : false;
 		let builderClass = '.modal.fade';
-		if (this._modalOptions.isFlyout) {
-			builderClass += '.flyout-dialog';
-		}
+		builderClass += this._modalOptions.dialogStyle === 'flyout' ? '.flyout-dialog'
+			: this._modalOptions.dialogStyle === 'callout' ? '.callout-dialog'
+				: this._modalOptions.dialogStyle === 'calloutCompact' ? '.callout-dialog.compact'
+					: '';
 
 		this._bodyContainer = DOM.$(`${builderClass}`, { role: 'dialog', 'aria-label': this._title });
 		const top = this.layoutService.offset?.top ?? 0;
 		this._bodyContainer.style.top = `${top}px`;
 		this._modalDialog = DOM.append(this._bodyContainer, DOM.$('.modal-dialog'));
-		this._modalContent = DOM.append(this._modalDialog, DOM.$('.modal-content'));
+
+		if (this._modalOptions.dialogStyle === 'callout' || this._modalOptions.dialogStyle === 'calloutCompact') {
+			let arrowClass = `.arrow.from-${this._modalOptions.dialogPosition}`;
+			this._modalContent = DOM.append(this._modalDialog, DOM.$(`.modal-content${arrowClass}`));
+
+			if (this._modalOptions.dialogPosition === 'below') {
+				if (hasMacClass) {
+					this._modalDialog.style.left = `${this._modalOptions.positionX - this._modalOptions.dialogXYOffset.xOffset}px`;
```

Comment:
The `hasMacClass` and `else` blocks here seem to have the same logic, what's the check for body.mac doing?

Reply:
Removed.